src/plugins/region_map/server/routes/opensearch.ts

```diff
@@ -59,10 +67,18 @@ export function registerGeospatialRoutes(router: IRouter) {
           index: schema.string(),
           size: schema.number(),
         }),
+        query: schema.maybe(schema.object({}, { unknowns: 'allow' })),
       },
     },
     async (context, req, res) => {
-      const client = context.core.opensearch.client.asCurrentUser;
+      let client;
+      // @ts-ignore
+      if (!req.query.dataSourceId) {
+        client = context.core.opensearch.client.asCurrentUser;
+      } else {
+        // @ts-ignore
+        client = await context.dataSource.opensearch.getClient(req.query.dataSourceId);
+      }
```

Comment:
can we extract this to be a util function for reuse? I saw 3 places are uses this